packages/driver/cypress/integration/e2e/multi-domain/multi_domain_cypress_api.spec.ts

```diff
@@ -0,0 +1,253 @@
+describe('multi-domain Cypress API', { experimentalSessionSupport: true, experimentalMultiDomain: true }, () => {
+  beforeEach(() => {
+    cy.visit('/fixtures/multi-domain.html')
+    cy.get('a[data-cy="multi-domain-secondary-link"]').click()
+  })
+
+  context('Commands', () => {
+    context('add', () => {
+      it('adds a custom command', () => {
+        cy.switchToDomain('foobar.com', () => {
+          // @ts-ignore
+          Cypress.Commands.add('foo', () => {
+            return cy.wrap('bar')
+          })
+
+          // @ts-ignore
+          cy.foo().then((shouldBeBar) => {
+            expect(shouldBeBar).to.equal('bar')
+          })
+        })
+      })
+
+      it('persists defined commands through spec bridge', () => {
+        cy.switchToDomain('foobar.com', () => {
+          // @ts-ignore
+          cy.foo().then((shouldBeBar) => {
+            expect(shouldBeBar).to.equal('bar')
+          })
+        })
+      })
+    })
+
+    context('overwrite', () => {
+      it('overwrites an existing command in the spec bridge', () => {
+        cy.switchToDomain('foobar.com', () => {
+          // @ts-ignore
+          Cypress.Commands.overwrite('foo', () => {
+            return cy.wrap('baz')
+          })
+
+          // @ts-ignore
+          cy.foo().then((shouldBeBaz) => {
+            expect(shouldBeBaz).to.equal('baz')
+          })
+        })
+      })
+
+      it('persists overwritten command through spec bridge', () => {
+        cy.switchToDomain('foobar.com', () => {
+          // @ts-ignore
+          cy.foo().then((shouldBeBaz) => {
+            expect(shouldBeBaz).to.equal('baz')
+          })
+        })
+      })
+    })
+  })
+
+  context('Keyboard', () => {
+    it('does NOT sync defaults', () => {
+      const defaults = Cypress.Keyboard.defaults({
+        keystrokeDelay: 30,
+      })
+
+      cy.switchToDomain('foobar.com', [defaults], ([primaryKeyboardDefaults]) => {
+        const multiDomainKeyboardDefaults = Cypress.Keyboard.defaults({})
+
+        expect(multiDomainKeyboardDefaults).to.not.deep.equal(primaryKeyboardDefaults)
+      })
+    })
+
+    it('allows a user to configure defaults', () => {
+      cy.switchToDomain('foobar.com', () => {
+        const multiDomainKeyboardDefaults = Cypress.Keyboard.defaults({
+          keystrokeDelay: 60,
+        })
+
+        expect(multiDomainKeyboardDefaults).to.deep.include({
+          keystrokeDelay: 60,
+        })
+      })
+    })
+
+    it('persists default configuration changes through spec bridge', () => {
+      cy.switchToDomain('foobar.com', () => {
+        const multiDomainKeyboardDefaults = Cypress.Keyboard.defaults({})
+
+        expect(multiDomainKeyboardDefaults).to.deep.include({
+          keystrokeDelay: 60,
+        })
+      })
+    })
+  })
+
+  context('Screenshot', () => {
+    it('does NOT sync defaults', () => {
+      Cypress.Screenshot.defaults({
```

Comment:
@mschile  should we move these tets to `multi_domain_screenshot.spec.ts`?

Reply:
I think an argument could be made either way but I think it's fine to leave them here since they follow the pattern of the other Cypress.* tests.